core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/DefaultParamInfo.scala

```diff
@@ -0,0 +1,121 @@
+  val StringArrayInfo = new ParamInfo[StringArrayParam]("list", "TypeConverters.toListString",
+    "as.array", "string[]")
+  val DoubleArrayInfo = new ParamInfo[DoubleArrayParam]("list", "TypeConverters.toListFloat",
+    "as.array", "double[]")
+  val IntArrayInfo = new ParamInfo[IntArrayParam]("list", "TypeConverters.toListInt",
+    "as.array", "int[]")
+  val ByteArrayInfo = new ParamInfo[ByteArrayParam]("list", "byte[]")
+  val DoubleArrayArrayInfo = new ParamInfo[DoubleArrayArrayParam]("object", "double[][]")
+  val StringStringMapInfo = new ParamInfo[StringStringMapParam]("dict", "Dictionary<string, string>")
+  val StringIntMapInfo = new ParamInfo[StringIntMapParam]("dict", "Dictionary<string, int>")
+  val ArrayMapInfo = new ParamInfo[ArrayMapParam]("object", "Dictionary<string, object>[]")
+  val TypedIntArrayInfo = new ParamInfo[TypedIntArrayParam]("object", "int[]")
+  val TypedDoubleArrayInfo = new ParamInfo[TypedDoubleArrayParam]("object", "double[]")
+  val UntypedArrayInfo = new ParamInfo[UntypedArrayParam]("object", "object[]")
+
+  val SeqTimeSeriesPointInfo = new ParamInfo[ServiceParam[_]]("object", "TimeSeriesPoint[]")
+  val SeqTargetInputInfo = new ParamInfo[ServiceParam[_]]("object", "TargetInput[]")
+  val SeqTextAndTranslationInfo = new ParamInfo[ServiceParam[_]]("object", "TextAndTranslation[]")
+  val DiagnosticsInfo = new ParamInfo[CognitiveServiceStructParam[_]]("object", "DiagnosticsInfo")
+  val TextAnalyzeTaskParamInfo = new ParamInfo[CognitiveServiceStructParam[_]]("object", "TAAnalyzeTask[]")
+
+  val UnknownInfo = new ParamInfo[Param[_]]("object", "object")
+
+  //noinspection ScalaStyle
+  def getGeneralParamInfo(dataType: Param[_]): ParamInfo[_] = {
+    dataType match {
+      case _: BooleanParam => BooleanInfo
+      case _: IntParam => IntInfo
+      case _: LongParam => LongInfo
+      case _: FloatParam => FloatInfo
+      case _: DoubleParam => DoubleInfo
+      case _: StringArrayParam => StringArrayInfo
+      case _: DoubleArrayParam => DoubleArrayInfo
+      case _: IntArrayParam => IntArrayInfo
+      case _: ByteArrayParam => ByteArrayInfo
+      case _: DoubleArrayArrayParam => DoubleArrayArrayInfo
+      case _: StringStringMapParam => StringStringMapInfo
+      case _: StringIntMapParam => StringIntMapInfo
+      case _: ArrayMapParam => ArrayMapInfo
+      case _: TypedIntArrayParam => TypedIntArrayInfo
+      case _: TypedDoubleArrayParam => TypedDoubleArrayInfo
+      case _: UntypedArrayParam => UntypedArrayInfo
+      case sp: ServiceParam[_] => getServiceParamInfo(sp)
+      case csp: CognitiveServiceStructParam[_] => getCognitiveServiceStructParamInfo(csp)
+      case cp: ComplexParam[_] => getComplexParamInfo(cp)
+      case p => {
+        logWarning(s"unsupported type $p")
+        UnknownInfo
+      }
+    }
+  }
+
+  //noinspection ScalaStyle
+  def getServiceParamInfo(dataType: ServiceParam[_]): ParamInfo[_] = {
```

Comment:
Int he future we might want to figure out a good way to abstract this stuff into "plugins" so that the core code is clean and each added parameter class has to do the legwork to hook into codegen/testgen